homeassistant/components/livisi/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for Livisi Home Assistant."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiolivisi import AioLivisi, errors as livisi_errors
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_HOST, CONF_PASSWORD, DOMAIN, LOGGER
+
+
+class LivisiFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Livisi Smart Home config flow."""
+
+    def __init__(self) -> None:
+        """Create the configuration file."""
+        self.host: str | None
+        self._aio_livisi: AioLivisi = None
+        self.data_schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=self.data_schema)
+
+        errors = {}
+        ip_password: dict[str, str] = user_input
+        try:
+            await self._login(ip_password)
+        except livisi_errors.WrongCredentialException:
+            errors["base"] = "wrong_password"
+        except livisi_errors.ShcUnreachableException:
+            errors["base"] = "shc_unreachable"
+        except livisi_errors.IncorrectIpAddressException:
+            errors["base"] = "wrong_ip_address"
+        else:
+            return await self.create_entity(ip_password)
+
+        return self.async_show_form(
+            step_id="user", data_schema=self.data_schema, errors=errors
+        )
+
+    async def _login(self, user_input: dict[str, str]) -> None:
+        """Login into Livisi Smart Home."""
+        self.host = user_input.get(CONF_HOST)
```

Comment:
The host is required in the form schema.
```suggestion
        self.host = user_input[CONF_HOST]
```

Reply:
It doesn't look like we need to store host as an instance attribute. We pass the user input when creating the entry.